disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+
+
+def check_url(url: str) -> bool:
+    """Check url
+
+    Parameters
+    ----------
+    url : str
+        url
+
+    Returns
+    -------
+    bool
+        True if url well formated, False if not well formated
+    """
+    regex = r"^(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)$"  # noqa: E501
+
+    if re.match(regex, url):
+        return True
+    else:
+        return False
+
+
+def download_file_from_url(url: str, folder_path: str) -> None:
+    """Download file
+
+    Parameters
+    ----------
+
+    url : str
+        URL of the file to download
+    file_name : str
+        Local name of the file
+    """
+
+    file_name = os.path.basename(url)
+
+    downloader = pooch.HTTPDownloader(progressbar=True)
+    pooch.retrieve(
+        url=url,
+        known_hash=None,
+        path=folder_path,
+        fname=file_name,
+        downloader=downloader,
+        progressbar=tqdm,
+    )
+
+
+# function to unzip one file into a folder
+def unzip_file(file_path: str, dest_path: str) -> None:
+    """Unzip a file into a folder
+
+    Parameters
+    ----------
+    file_path : str
+        Path of the file to unzip
+    dest_path : str
+        Path of the destination folder
+    """
+
+    with zipfile.ZipFile(file_path, "r") as zip_ref:
+        zip_ref.extractall(dest_path)
+
+
+def download_file_content(url_local_path: tuple, overwrite: bool = False, unzip: bool = False) -> None:
+    """Download the files based on a tuple (local path , url).
+    Unzip the file if it is a zip file and if requested.
+
+    Parameters
+    ----------
+    file_path : tuple
+        Tuple containing the local file path and the url
+    overwrite : bool, optional
+        Overwrite the raw data file is already existing, by default False
+    unzip : bool, optional
+        Unzip the file if it is a zip file, by default False
+    """
+
+    local_path, url = url_local_path
+    if check_url(url):
+        # create station folder if not existing
+        if not os.path.exists(local_path):
+            os.makedirs(local_path)
+        # download the file
+        download_file_from_url(url, local_path)
+        print(f"Download {url} into {local_path}")
+        if url.endswith(".zip") and unzip:
+            zip_local_path = os.path.join(local_path, os.path.basename(url))
+            unzip_file(zip_local_path, local_path)
+            os.remove(zip_local_path)
+
+    else:
+        # raise an error if url not well formated
+        raise ValueError(f"URL {url} is not well formated. Please check the URL.")
+
+
+def download_all_files(
+    raw_folder: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+    overwrite: bool = False,
+    unzip: bool = False,
+):
+    """Batch function to get all yaml files that coantain
+    the 'data_url' key and download the data locally.
+
+    Parameters
+    ----------
+    raw_folder : str, optional

```

Comment:
Let's use `disdrodb_dir` and add "Raw" internally.
`raw_dir` and `processed_dir` are only only used in internal L0 processing.

Reply:
ok